test/e2e/suite/secrettemplate/secrettemplate.go

```diff
@@ -0,0 +1,469 @@
+		Expect(secret.Annotations).To(HaveKeyWithValue("foo", "bar"))
+		Expect(secret.Annotations).To(HaveKeyWithValue("bar", "foo"))
+		Expect(secret.Labels).To(HaveKeyWithValue("abc", "123"))
+		Expect(secret.Labels).To(HaveKeyWithValue("def", "456"))
+
+		By("adding Annotations and Labels to SecretTemplate should appear on the Secret")
+
+		crt.Spec.SecretTemplate.Annotations["random"] = "annotation"
+		crt.Spec.SecretTemplate.Annotations["another"] = "random annotation"
+		crt.Spec.SecretTemplate.Labels["hello"] = "world"
+		crt.Spec.SecretTemplate.Labels["random"] = "label"
+
+		crt, err = f.CertManagerClientSet.CertmanagerV1().Certificates(f.Namespace.Name).Update(context.Background(), crt, metav1.UpdateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() map[string]string {
+			secret, err = f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).Get(context.Background(), secretName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			return secret.Annotations
+		}, "5s", "1s").Should(HaveKeyWithValue("random", "annotation"))
+		Expect(secret.Annotations).To(HaveKeyWithValue("bar", "foo"))
+		Expect(secret.Annotations).To(HaveKeyWithValue("foo", "bar"))
+		Expect(secret.Annotations).To(HaveKeyWithValue("another", "random annotation"))
+
+		Eventually(func() map[string]string {
+			secret, err = f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).Get(context.Background(), secretName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			return secret.Labels
+		}, "5s", "1s").Should(HaveKeyWithValue("hello", "world"))
+		Expect(secret.Labels).To(HaveKeyWithValue("def", "456"))
+		Expect(secret.Labels).To(HaveKeyWithValue("abc", "123"))
+		Expect(secret.Labels).To(HaveKeyWithValue("random", "label"))
+
+		By("removing Annotations and Labels in SecretTemplate should get removed on the Secret")
+
+		crt, err = f.CertManagerClientSet.CertmanagerV1().Certificates(f.Namespace.Name).Get(context.Background(), crt.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		delete(crt.Spec.SecretTemplate.Annotations, "foo")
+		delete(crt.Spec.SecretTemplate.Annotations, "random")
+		delete(crt.Spec.SecretTemplate.Labels, "abc")
+		delete(crt.Spec.SecretTemplate.Labels, "another")
+
+		crt, err = f.CertManagerClientSet.CertmanagerV1().Certificates(f.Namespace.Name).Update(context.Background(), crt, metav1.UpdateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() map[string]string {
+			secret, err = f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).Get(context.Background(), secretName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			return secret.Annotations
+		}, "5s", "1s").ShouldNot(HaveKey("foo"))
+
+		Expect(secret.Annotations).ToNot(HaveKey("random"))
+		Expect(secret.Labels).ToNot(HaveKey("abc"))
+		Expect(secret.Labels).ToNot(HaveKey("another"))
+	})
+
+	it("should update the values of keys that have been modified in the SecretTemplate", func() {
```

Comment:
It seems like this test timeout is too short: https://github.com/jetstack/cert-manager/issues/4682

Reply:
Fixed the tests so am resolving.